packages/form-builder/src/containers/HeaderContainer.jsx

```diff
@@ -24,7 +24,8 @@ const HeaderContainer = ({
         setSaving(true);
         Storage.save({blocks, formSettings})
             .catch((error) => alert(error.message))
-            .finally(() => {
+            .then((updatedSettings) => {
+                dispatch(setFormSettings({pageSlug: updatedSettings.pageSlug}));
```

Comment:
@jonwaldstein This is where we are updating the page slug in the builder, as it may have been updated on the server (for uniqueness).